license_copyright/license.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Update copyright license'
+on:
+  schedule:
+    - cron: '0 0 1 1 *'
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update the License
+        run: |
+          (echo "Copyright (C) 2016-$(date +"%Y") TheAlgorithms and contributors";  tail -n +2 LICENSE) > License.tmp
+          mv License.tmp LICENSE
+      - name: Commit and push changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        id: commit-push
+        with:
+          commit_message: "docs: updating `License` copyright"
+          branch: "license-${{ github.sha }}"
+          create_branch: true
+      - name: Creating and merging the PR
+        shell: bash
+        if: steps.commit-push.outputs.changes_detected == 'true'
+        run: |
+          gh pr create --base ${GITHUB_REF##*/} --head license-${{ github.sha }} --title 'docs: updating `License` copyright' --body 'Updated License copyright (see the diff. for changes).'
+        env:
+          GH_TOKEN: ${{ github.token }}
```

Comment:
```suggestion
name: 'Update copyright license'
description: Update license years automatically
author: "TheAlgorithms"
inputs:
  filename:
    description: License file name
    required: true
    default: LICENSE
runs:
  using: composite
  steps:
    - name: Update the License
      run: |
        (echo "Copyright (C) 2016-$(date +"%Y") TheAlgorithms and contributors";  tail -n +2 ${{ inputs.filename }}) > License.tmp
        mv License.tmp ${{ inputs.filename }}
    - name: Setup Git configurations
      shell: bash
      run: |
        git config --global user.name github-actions[bot]
        git config --global user.email 'github-actions@users.noreply.github.com'
    - name: Commit to a new branch and push changes
      shell: bash
      run: |
        git checkout -b license-update-${{ github.sha }}
        git commit -m "docs: updating Copyright license year"
        git push
    - name: Creating and merging the PR
      shell: bash
      if: steps.commit-push.outputs.changes_detected == 'true'
      run: |
        gh pr create --base master --head license-update-${{ github.sha }} --title 'docs: updating `License` copyright' --body 'Updated License copyright (see the diff. for changes).'
      env:
        GH_TOKEN: ${{ github.token }}

```

Reply:
I'm not sure if line 29 will work as the `commit-push` was a step defined in the other workflow.

Reply:
Ah yes, thanks for noticing! A similar approach might work with `git diff --no-patch --exit-code` should do the trick I think.

Reply:
Thank you!